### PR TITLE
video: New experimental module

### PR DESCRIPTION
**Pull request by ibgreen**
For now this provide a minimal browser-only video loader. 

@chrisgervang  This module is intended to be a work area for integrating video functionality from hubble.gl

### CODE REVIEW

**Comment by ibgreen** [comment]
> Awesome! Thanks for this. Did you mean to copy over all the test data files? (Maybe you could help me understand their purpose?)

Good catch. Removed. They were copy pasta from the images module. 

We do however need some test data to land this.

**Comment by chrisgervang** [comment]
A small video to load that has an open license? Which format(s), mp4 or
webm?

**Comment by ibgreen** [comment]
- Right now we load using browser's video element, so a range of formats supported by browsers, including those two.
- Not sure if video element can consume GIF...
- The test files have got to be small (ideally < 1MB each), we don't want to bloat the loaders.gl repo with 100 megabyte test files.

**Comment by ibgreen** [comment]
@chrisgervang Added `GIFBuilder` class (based on gifshot)

File: dev-docs/RFCs/vNext/encoding-rfc.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+## Encoder APIs
+
+## encode
+
+Simple atomic encode should reverse the effects of load.
+
+```js
+const {load, encode, write} from '@loaders.gl/core`;
+const parsedBatchIterator = load(filename, ShapefileLoader);
+const encodedBatchIterator = encode(parsedBatchIterator, KMLEncoder);
```
Comment (chrisgervang) [inline dev-docs/RFCs/vNext/encoding-rfc.md:22]: `const encodedBatch`?

File: modules/video/src/video-loader.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// __VERSION__ is injected by babel-plugin-version-inline
+/* global __VERSION__ */
+import parseVideo from './lib/parsers/parse-video';
+
+// @ts-ignore TS2304: Cannot find name '__VERSION__'.
+const VERSION = typeof __VERSION__ !== 'undefined' ? __VERSION__ : 'latest';
+
+const EXTENSIONS = ['mp4'];
```
Comment (chrisgervang) [inline modules/video/src/video-loader.js:8]: This probably can support more extensions, like `webm`. We can experiment later.

**Comment by ibgreen** [comment]
Will address comments separately.

**Pull request merged**